code/modules/antagonists/revolution/revolution.dm

```diff
@@ -455,6 +455,9 @@
 			to_chat(headrev_mind.current, span_hear("You hear something crackle in your ears for a moment before a voice speaks. \
 				\"Please stand by for a message from your benefactor. Message as follows, provocateur. \
 				<b>You have been chosen out of your fellow provocateurs to rename the station. Choose wisely.</b> Message ends.\""))
+		for (var/obj/machinery/taxed_machine as anything in GLOB.production_machines)
+			qdel(taxed_machine.GetComponent(/datum/component/payment))
+
```

Comment:
So this is a bit esoteric but
is there any reason the payment component shouldn't self remove on rev win from everything?
if so, could you make removing on win a flag or such that gets passed into the component rather then doing it manually like this?

In general I'd prefer for everything a component does post addition to be tied to a signal. Reacting to events and such.